src/ArrayValueCalculator.java

```diff
@@ -0,0 +1,33 @@
+public class ArrayValueCalculator {
+
+    public static String doCalc(String stringArray[][]) throws ArraySizeException, ArrayDataException {
+        int sum = 0;
+
+        for (int i = 0; i < stringArray.length; i++) {
+            for (int j = 0; j < stringArray[i].length; j++) {
+                if (i <= 0 || j <= 0) {
+                    throw new IllegalArgumentException("Each element in the array must be non-negative and greater than zero");
+                }
+            }
+        }
+
+            if (stringArray == null) {
+                throw new NullPointerException("The array must not be null, the array must be 4x4");
+            }
+            if (stringArray.length != 4 || stringArray[0].length != 4) {
```

Comment:
`stringArray.length != 4` <- this is correct. No questions.
But this -> `stringArray[0].length != 4` is not correct.
You've started in the correct way, but haven't finished the validation. You're validating only 1 row here, but it's required to validate them all because the matrix must be of size 4x4 according to the description.